### ISSUE TITLE
执行train_pretrain_vlm.py代码训练的时候，loss损失值比较高。保持在7.5左右，什么原因？

### ISSUE THREAD

**Issue by ctrlZ333** [jingyaogong/minimind-v#75]
![Image](https://github.com/user-attachments/assets/0aa8fac3-d6f7-4ae7-8a4d-015865575497)

**Comment by TechnetiumMan** [comment]
并且我发现，即使直接载入他训好了的 pretrain_vlm_768.pth 或 sft_vlm_768.pth 接着训，loss也是7.几i

**Comment by wwwxuzhenyu** [comment]
同问。我也是 loss 收敛到 7.3 左右就不再下降了，eval 出来完全不说人话。

**Comment by TechnetiumMan** [comment]
似乎问题出在 trainer/train_pretrain_vlm.py line 107
model.load_state_dict(state_dict, strict=False) 这一句，根本没有load进去任何东西，整个模型都是随机初始化的，没有任何的语言能力

**Comment by wwwxuzhenyu** [comment]
是的。checkpoint 里面的名字和实际模型参数的名字都不对应，导致参数变成随机初始化的了

**Comment by wwwxuzhenyu** [comment]
https://huggingface.co/jingyaogong/MiniMind2-V/blob/main/pytorch_model.bin 从这个 ckpt 加载似乎可以

**Comment by TechnetiumMan** [comment]
> https://huggingface.co/jingyaogong/MiniMind2-V/blob/main/pytorch_model.bin 从这个 ckpt 加载似乎可以

这个是可以，但是已经训好了，不能从头开始训了
从头开始训所需要的minimind语言模型，版本和minimind-v对不上

**Comment by TechnetiumMan** [comment]
等作者更新可下载的minimind语言模型版本吧

**Comment by wokaikaixinxin** [comment]
同样的问题，运行 train_pretrain_vlm.py，损失函数一直是7.5左右，不收敛。恳请作者更新代码和权重，多谢

![Image](https://github.com/user-attachments/assets/1c263995-6050-4788-b610-6978dde6b1f3)

**Comment by wokaikaixinxin** [comment]
其实是作者的readme没有及时更新。去纯语言minimind中找新的权重。
参考解决方法：
1）下载新的纯语言minimind权重
```python
cd /root/minimind-v/
mkdir out
cd out
wget https://hf-mirror.com/jingyaogong/MiniMind2-Pytorch/resolve/main/full_sft_512.pth
```
2）修改`trainer/train_pretrain_vlm.py`第104行
```
ckp = f'{args.save_dir}/full_sft_{model_config.hidden_size}{moe_path}.pth'
```
3）好了

**Comment by HeXioahei** [comment]
> 其实是作者的readme没有及时更新。去纯语言minimind中找新的权重。 参考解决方法： 1）下载新的纯语言minimind权重
> 
> cd /root/minimind-v/
> mkdir out
> cd out
> wget https://hf-mirror.com/jingyaogong/MiniMind2-Pytorch/resolve/main/full_sft_512.pth
> 2）修改`trainer/train_pretrain_vlm.py`第104行，即修改路径名称
> 
> ```
> ckp = f'{args.save_dir}/full_sft_{model_config.hidden_size}{moe_path}.pth'
> ```
> 
> 3）2卡条件下，前面轮次的loss数值是3.3左右。不知道损失数值初始3.3正常吗。训练4个epoch，loss都是3左右。大家复现的loss都是多少？
> 
> ![Image](https://github.com/user-attachments/assets/fc8f2722-85d7-44ba-b557-80488d0d3fd4)
> 
> ![Image](https://github.com/user-attachments/assets/94cb22c3-1386-4ab5-89b2-9fe6c7c7b174)

我用你的方法试了一下，loss也是在3左右。不过，你这个权重对应的应该是之前minimind的吧，不是minimind-v的吧。根据库的名字来看，minimind的权重库是https://hf-mirror.com/jingyaogong/MiniMind2-Pytorch/tree/main，若将这里的权重用于minimind-v，loss反而比较小，一直是3点多；minimind-v的权重库是https://huggingface.co/jingyaogong/MiniMind2-V-PyTorch/tree/main，若将这里的权重用于minimind-v，loss反而比较大，一直是7点多。不过我还是没搞懂问题到底出在哪。

**Comment by cqtju** [comment]
> 这个问题确实是因为几个月来的`pth`是旧版本的kv，导致语言模型初始参数没有注入进来；已经重新上传解决 😊

看起来只更新了huggingface的权重， 没有更新modelscope里的权重
